### PR TITLE
[GLUTEN-5001][CORE] Support mapping to native function for HiveGenericUDF

### PR DESCRIPTION
**Pull request by WangGuangxin**
## What changes were proposed in this pull request?
Support HiveGenericUDF when mapping to native function

(Fixes: \#5001)

### CODE REVIEW

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveUDFTransformer.scala [hunk 1]
```diff
@@ -22,34 +22,37 @@ import org.apache.spark.sql.catalyst.expressions._
...
       attributeSeq: Seq[Attribute]): ExpressionTransformer = {
-    if (!isHiveSimpleUDF(expr)) {
-      throw new UnsupportedOperationException(s"Expression $expr is not a HiveSimpleUDF")
+    val udfName = expr match {
+      case s: HiveSimpleUDF =>
+        s.name.stripPrefix("default.")
+      case g: HiveGenericUDF =>
+        g.name.stripPrefix("default.")
+      case _ =>
+        throw new UnsupportedOperationException(
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveUDFTransformer.scala:42]: Let's use `GlutenNotSupportException`, which is newly introduced for expected fallback.

**Review by PHILO-HE — COMMENTED** [review]
Please help address one minor comment. Thank!
@taiyang-li, could you review this pr?

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveUDFTransformer.scala (hunk 1 above)
Comment (WangGuangxin) [reply gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveUDFTransformer.scala]: Thanks for review. Fixed

**Comment by taiyang-li** [comment]
LGTM

**Review by taiyang-li — APPROVED** [review]
LGTM

**Review by zhouyuan — COMMENTED** [review]
looks good to me. cc @marin-ma

**Pull request merged**